core/common/src/Localization.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Localization
+ */
+
+import { Logger } from "@bentley/bentleyjs-core";
+
+ /** @public */
+ // This should be more clearly defined or removed as we remove I18N and its dependencies (Formerly i18next::TranslationOptions).
+ type LocalizationOptions = any;
+
+/** The interface defining the localization requirements of iModelJs.
+ * @public
+ */
+export interface LocalizationClient {
+  getLocalizedString(key: string | string[], options?: LocalizationOptions): string;
+  getLocalizedStringWithNamespace(namespace: string, key: string | string[], options?: LocalizationOptions): string;
+  getEnglishString(namespace: string, key: string | string[], options?: LocalizationOptions): string;
```

Comment:
Why are we giving special treatment to English?

Reply:
Good question - we should probably remove this from the localization API. It was included in this PR to support all previous usage of I18N. As of now the only consumer of getEnglishString is core/frontend/src/tools/Tool.ts, in _englishKeyin()_.

Reply:
Yes, I think the reason that's used is so the key-in browser/system always uses the English strings even if strings are localized, sort of a like an invariant name. Is that true @pmconne and if not can should we make that true?

Reply:
We search first for localized keyin; if none is found we fall back to English keyin. It was Barry's idea to localize keyin strings; @bbastings and I had some reservations, so this was the compromise.

Reply:
Is there a reason this needs to be in the public interface?  Maybe it should be in a separate, internal interface?  If it needs to be here, maybe it can be renamed to `getKeyinString`?

Reply:
Not going to hold the PR for this.